lib/ip_access_control/options.ex

```diff
@@ -122,7 +122,9 @@ defmodule IpAccessControl.Options do
   end
 
   defp evaluate(:allow, allow_list) do
-    IpAccessControl.parse_allow_list(allow_list)
+    allow_list
+    |> IpAccessControl.parse_allow_list()
+    |> BitwiseIp.Blocks.optimize()
```

Comment:
The optimization pass is nice, but in my case only applicable for pre-compiled (e.g., non-runtime) allow lists. :(

Reply:
Yeah, you almost certainly don't want to call `BitwiseIp.Blocks.optimize/1` at runtime anyway. Except in pathological cases, I can't imagine the improvement to `BitwiseIp.Blocks.contain?/2` made by merging things together outweighs the cost of actually doing the optimization. It's an iteration til fixpoint that repeatedly does an O(n<sup>2</sup>) scan over the list looking for blocks to merge. I'm not sure the juice is worth the squeeze. It's sort of just a parlor trick I cribbed from [cider](https://github.com/IanLuites/cider/blob/e309901c8f5fb5fc3e810f4540b5cfcb1d8a51c1/lib/cider.ex#L482) for the fun of it.